### DISCUSSION TITLE
When I use BuildSessionFactory() or inject the Controller constructor,it took 30s.can u help me?thanks

### DISCUSSION THREAD

**Discussion by gzj132** [nhibernate/nhibernate-core#3249] [General] [answered]
hi:
this is my code 
![8A10B430-91D8-44AC-999B-AB4873475AE8](https://user-images.githubusercontent.com/50973061/223335054-912254e2-e595-401f-a986-9b018576bd92.jpeg)

**Comment by oskarb** [comment, marked as answer]
30s is a classical TCP timeout scale. Maybe you have some host name or IP in your configuration that doesn't respond.

**Reply by gzj132** [reply]
i examined the source code，and i found that running conn.open（） took a lot of time
![Uploading E5A8EEB1-93C0-4CE6-8386-102068F99C70.jpeg…]()

**Reply by gzj132** [reply]
![78712D1C-0B04-422C-B9FA-4C167E73DD38](https://user-images.githubusercontent.com/50973061/223377823-57880347-aa42-4e72-abc9-d2b1ac8d5bab.jpeg)

**Reply by gzj132** [reply]
hi，oskarb，i'm glad u can help me。
is there a congifuration error in my connection database file
![6C4468D4-BF41-42BA-A26C-0B45F3E00739](https://user-images.githubusercontent.com/50973061/223378937-07139be5-3c78-4417-96e3-0610cd712b2f.jpeg)

**Reply by oskarb** [reply]
As I said, my suspicion is that you have some kind of network issue.

If `conn.Open()` returns after about 30s with no exception, one idea is that you server has multiple IP addresses. Perhaps your server has both IPv4 and IPv6 but your IPv6 network is broken so it doesn't actually work? Or there are multiple IPv4 configured in DNS but only one of them works. This type of situation could lead to delays as the client tries them one at a time.

**Reply by gzj132** [reply]
First I would like to thank you, I thought about this reason, but not because of this, I happened to think about ssl, so I tried to add sslmode=none after the connection string, I solved the problem, wish you all the best

**Reply by oskarb** [reply]
Running without encryption feels more like a workaround, not a solution. If the server doesn't support SSL currently, I guess that is the only thing you can do.

**Reply by gzj132** [reply]
u r right
